force-app/main/default/classes/OpportunityTriggerHandler.cls

```diff
@@ -0,0 +1,180 @@
+/*
+ * OpportunityTriggerHandler extends the parent class TriggerHandler to specify actions for Opportunities.
+ * 
+ * Included methods:
+ * 
+ *  - beforeInsert: Sets a default Type for all Opportunities created without one
+ *  - beforeUpdate: Validates if the Amount is under a certain threshhold, upends Stage changes to the Opportunity Description, and sets a Primary Contact if none is provided
+ *  - beforeDelete: Validates that Opportunities are not Closed before deleting
+ *                  *** IMPORTANT NOTE: If this method is changed, review requirements for not allowing deletion of Closed Opportunities for
+ *                                      Accounts with Industry of "Banking" ***
+ *  - afterInsert: Creates a Task for each new Opportunity
+ *  - afterDelete: Alerts owner of the Oppotunity when it is deleted
+ *  - afterUndelete: Reassigns undeleted Opportunities to a new Primary Contact
+ * 
+ * @author Ilana Shotkin
+*/
+
+public with sharing class OpportunityTriggerHandler extends TriggerHandler {
+
+    // Class variables
+    private List<Opportunity> newOppList;
+    private List<Opportunity> oldOppList;
+    private Map<Id, Opportunity> newOppMap;
+    private Map<Id, Opportunity> oldOppMap;
+
+    // Class Constructor
+    public OpportunityTriggerHandler(List<Opportunity> newOppList, List<Opportunity> oldOppList, Map<Id, Opportunity> newOppMap, Map<Id, Opportunity> oldOppMap) {
+        this.newOppList = newOppList;
+        this.oldOppList = oldOppList;
+        this.newOppMap = newOppMap;
+        this.oldOppMap = oldOppMap;
```

Comment:
Run from a Trigger context, you would still have access to the Trigger. variables here. You would not NEED to pass them. However, if you wanted manually use these same methods with a batch of records, maybe you turn off the triggers and want to process them separately or something, I could see this possibility being useful. I more just wanted to point out that you can use Trigger.new... here as well.